rnm: add page (#7603)
tldr-pages/tldr: pages/common/rnm.md

# rnm

> Batch renaming tool.
> More information: <https://github.com/neurobin/rnm>.

> - Rename a file adding a specific text to the filename:

Reviewer: How doe  it add it to the filename? Does it append, prepend, or do something else?

Author: The '/fn/' is for the full file name including extension ('/f/' is without extension), so in this example it would append.
From the man page:
```
  FILENAME
       1. /fn/ : Full name of the file.

       2. /n/ : File name without extension.

       3. /e/ : File extension.

       4. /rn/ : Replaced Name, generated by replace strings.

       5. /pd/ : Parent directory name of the current file or directory.

       6. /wd/ : Current working directory name.

       7. /nsf/: Name (or name string) from name string file.
```

`rnm -ns '/fn/{{text_to_add}}' {{path/to/file}}`

> - Rename all files in the current directory appending an index to the filename:

Reviewer: Is this before or after the file extension?

Author: This would be after the file extension. Maybe it would be better to change it to '/n/', since that's probably what most people would use.

`rnm -ns '{{/n/ /i/./e/}}' {{./*}}`

> - Append index with leading zeroes and with starting index 4:

Reviewer: Perhaps this will read better?
```suggestion
- Append an index with leading zeroes, starting at an index of 4:
```

`rnm -ns '/n/ /id/./e/' -ifl 3 -inc 1 -si 4 ./*`

- Rename all files in the current directory replacing underscores with spaces ('/regex/replace/modifier'):

> `rnm -rs '{{/_/ /g}}' {{./*}}`

Reviewer: Perhaps a more generic example here would help aid clarity?

Author: What would you suggest?

Reviewer: i.e. another example before that one that better demonstrates the basic syntax?

# rnm

> > Batch renaming tool.
> 

Reviewer 2: ```suggestion
> Batch renaming tool.
```

> > Batch renaming tool.  

Author: Added two spaces at the end. Don't really know what else to do.

Author: In response of: https://github.com/tldr-pages/tldr/pull/7603#issuecomment-1003717302

Reviewer 3: The problem is the empty line between this line and the more information line. Plus, there is no need to put trailing spaces in any end of lines.

> > More information: https://github.com/neurobin/rnm

Reviewer 2: ```suggestion
> More information: <https://github.com/neurobin/rnm>.
```

> > Batch renaming tool.  
> 
> > More information: <https://github.com/neurobin/rnm>.

Reviewer 3: ```suggestion
> Batch renaming tool. 
> More information: <https://github.com/neurobin/rnm>.
```

> - Rename by adding something to oldname:

Reviewer 2: Perhaps this better
```suggestion
- Rename a file adding a specific text to the filename:
```

> `rnm -ns '/fn/ some text to add' oldfile`

Reviewer 2: ```suggestion
`rnm -ns '/fn/{{text_to_add}}' {{path/to/file}}`
```

> - Rename all by appending index to file names:

Reviewer 2: ```suggestion
- Rename all files in the current directory appending an index to the filename:
```

> `rnm -ns '/n/ /i/./e/' ./*`

Reviewer 2: ```suggestion
`rnm -ns '{{/n/ /i/./e/}}' {{./*}}`
```

- Append index with leading zeroes and with starting index 4:

`rnm -ns '/n/ /id/./e/' -ifl 3 -inc 1 -si 4 ./*`

> - Change underscores to spaces in file name ('/regex/replace/modifier'):

Reviewer 2: ```suggestion
- Rename all files in the current directory replacing underscores with spaces ('/regex/replace/modifier'):
```

> `rnm -rs '/_/ /g' ./*`

Reviewer 2: ```suggestion
`rnm -rs '{{/_/ /g}}' {{./*}}`
```
